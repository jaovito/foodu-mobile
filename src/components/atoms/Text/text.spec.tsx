import {render} from '@test-utils';
import {Text} from '.';

describe('Text Tests', () => {
  const textStr = 'John doe';

  it('should render correctly', () => {
    const text = render(<Text>{textStr}</Text>);

    expect(text).toBeDefined();
  });

  it('should be able to find the text string inside component', async () => {
    const {findByText} = render(<Text>{textStr}</Text>);

    const text = await findByText(textStr);

    expect(text).toBeTruthy();
  });

  it('should not be able to find the text string inside component', async () => {
    const {queryByText} = render(<Text>{textStr}</Text>);

    const text = queryByText('potato');

    expect(text).toBeNull();
  });

  it('should match with snapshot', () => {
    const text = render(<Text>{textStr}</Text>);

    expect(text).toMatchSnapshot();
  });
});
