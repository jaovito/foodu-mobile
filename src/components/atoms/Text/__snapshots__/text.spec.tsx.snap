// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Tests should match with snapshot 1`] = `
<Text
  style={
    [
      {
        "color": "#0D0D0D",
        "fontFamily": "Poppins_400Regular",
        "fontSize": 31,
      },
    ]
  }
>
  John doe
</Text>
`;
